cmd/pipeline/deploy.go

```diff
@@ -249,6 +254,26 @@ func (pc *Command) ExecuteDeployPipeline(ctx context.Context, opts *DeployOption
 	return nil
 }
 
+// shouldRedeployDependencies determines if the pipeline should redeploy dependencies
+// default behavior is set by cluster config, but can be overridden by the user using the flag --dependencies
+func shouldRedeployDependencies(opts *DeployOptions) bool {
+	isForceRedeployDependenciesSetInOktetoInstance := env.LoadBoolean(constants.OktetoForceRedeployDependencies)
+	isInsideDependency := env.LoadBoolean(constants.OktetoIsDependencyEnvVar)
+	if !isInsideDependency && isForceRedeployDependenciesSetInOktetoInstance {
+		// the user forces --dependencies=false
+		if opts.DependenciesIsSet && !opts.RedeployDependencies {
+			return false
+		}
+		return true
+	}
+
+	if isInsideDependency {
```

Comment:
Same about moving this condition first in the function, so condition in line `262` would be simpler